### PR TITLE
PE adjustments

### PR DESCRIPTION
**Pull request by segraef**
## Description
Currently testing all multi-PE services as per https://learn.microsoft.com/en-us/azure/private-link/private-endpoint-overview#private-link-resource

Private-link resource name | Resource type | Sub-resources
-- | -- | --
Azure Automation | Microsoft.Automation/automationAccounts | Webhook, DSCAndHybridWorker
Azure Backup | Microsoft.RecoveryServices/vaults | AzureBackup, AzureSiteRecovery
Azure Batch | Microsoft.Batch/batchAccounts | batchAccount, nodeManagement
Azure Cosmos DB | Microsoft.AzureCosmosDB/databaseAccounts | SQL, MongoDB, Cassandra, Gremlin, Table
Azure Databricks | Microsoft.Databricks/workspaces | databricks_ui_api, browser_authentication
Azure Media Services | Microsoft.Media/mediaservices | keydelivery, liveevent, streamingendpoint
Azure Storage | Microsoft.Storage/storageAccounts | Blob (blob, blob_secondary)Table (table, table_secondary)Queue (queue, queue_secondary)File (file, file_secondary)Web (web, web_secondary)Dfs (dfs, dfs_secondary)
Azure Synapse Analytics | Microsoft.Synapse/workspaces | Sql, SqlOnDemand, Dev

Closes
- https://github.com/Azure/Azure-Verified-Modules/issues/620
- https://github.com/Azure/bicep-registry-modules/issues/962
- https://github.com/Azure/bicep-registry-modules/issues/946
- https://github.com/Azure/bicep-registry-modules/issues/1042

AVM Issues
- https://github.com/Azure/Azure-Verified-Modules/pull/621

## Pipeline Reference

| Pipeline |
| -------- |
| [avm.res.automation.automation-account](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.automation.automation-account.yml)     |
| [avm.res.batch.batch-account](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.batch.batch-account.yml)|
| [avm.res.databricks.workspace](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.databricks.workspace.yml)|
| [avm.res.document-db.database-account](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.document-db.database-account.yml)|
| [avm.res.storage.storage-account](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.storage.storage-account.yml)|
| [avm.res.synapse.workspace](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.synapse.workspace.yml) |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/batch/batch-account/main.bicep [hunk 1]
```diff
@@ -204,18 +204,30 @@ resource batchAccount_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@
...
-    privateLinkServiceConnections: [
+    name: privateEndpoint.?name ?? 'pep-${last(split(batchAccount.id, '/'))}-${privateEndpoint.service}-${index}'
+    privateLinkServiceConnections: privateEndpoint.?manualPrivateLinkServiceConnections != true ? [
       {
-        name: name
+        name: privateEndpoint.?privateLinkServiceConnectionName ?? '${last(split(batchAccount.id, '/'))}-${privateEndpoint.service}-${index}'
         properties: {
           privateLinkServiceId: batchAccount.id
           groupIds: [
-            privateEndpoint.?service ?? 'batchAccount'
```
Comment (AlexanderSehr) [inline avm/res/batch/batch-account/main.bicep:213]: Why did we remove the default for this and other resources?
Modules with only one sub-resource type should implement schema 1 (with a default)

Comment (segraef) [reply avm/res/batch/batch-account/main.bicep:213]: These are all services with more than one sub-resource (see table) which means they need to align to schema 2.

File: avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep [hunk 2]
```diff
@@ -94,7 +94,7 @@ resource keyPermissions 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
   name: guid('msi-${keyVault::key.id}-${location}-${managedIdentity.id}-Key-Key-Vault-Crypto-User-RoleAssignment')
   scope: keyVault::key
   properties: {
-    principalId: '711330f9-cfad-4b10-a462-d82faa92027d' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
+    principalId: '09ef9f59-e7b6-422e-84e4-508cfc8d64e7' // '711330f9-cfad-4b10-a462-d82faa92027d' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
     roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '12338af0-0e69-4776-bea7-57ae8d297424') // Key Vault Crypto User
```
Comment (segraef) [inline avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep:98]: Needs to be reverted before merging into main.

File: avm/res/databricks/workspace/tests/e2e/max/dependencies.bicep [hunk 3]
```diff
@@ -94,7 +94,7 @@ resource keyPermissions 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
   name: guid('msi-${keyVault::key.id}-${location}-${managedIdentity.id}-Key-Key-Vault-Crypto-User-RoleAssignment')
   scope: keyVault::key
   properties: {
-    principalId: '711330f9-cfad-4b10-a462-d82faa92027d' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
+    principalId: '09ef9f59-e7b6-422e-84e4-508cfc8d64e7' // '711330f9-cfad-4b10-a462-d82faa92027d' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
```
Comment (segraef) [inline avm/res/databricks/workspace/tests/e2e/max/dependencies.bicep:97]: Needs to be reverted before merging into main.

**Review by AlexanderSehr — APPROVED** [review]
Merge to feature branch

File: avm/res/databricks/workspace/tests/e2e/max/dependencies.bicep (hunk 3 above)
Comment (AlexanderSehr) [reply avm/res/databricks/workspace/tests/e2e/max/dependencies.bicep:97]: fixed in target

File: avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep (hunk 2 above)
Comment (AlexanderSehr) [reply avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep:98]: fixed in target

File: avm/res/batch/batch-account/main.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply avm/res/batch/batch-account/main.bicep:213]: ah wow, ok, thanks :)

**Pull request merged**